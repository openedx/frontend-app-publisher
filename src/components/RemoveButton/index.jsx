import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Close } from '@edx/paragon/icons';

const RemoveButton = ({
  className,
  label,
  onRemove,
  targetFieldNumber,
}) => (
  <button
    type="button"
    className={classNames('close float-none d-flex justify-content-center align-items-center', className)}
    aria-label={label}
    title={label}
    onClick={() => onRemove(targetFieldNumber)}
  >
    <Close
      id="remove-field"
    />
  </button>
);

RemoveButton.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired,
  targetFieldNumber: PropTypes.number.isRequired,
};

RemoveButton.defaultProps = {
  className: '',
};

export default RemoveButton;
