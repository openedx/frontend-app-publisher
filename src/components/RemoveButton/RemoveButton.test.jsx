import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RemoveButton from './index';

const mockOnRemove = jest.fn(targetFieldNumber => targetFieldNumber);

const defaultProps = {
  label: 'Test Label',
  onRemove: mockOnRemove,
  targetFieldNumber: 0,
};

describe('RemoveButton', () => {
  it('renders correctly', () => {
    const component = shallow(<RemoveButton {...defaultProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('calls the `onRemove` function with the target field number when the button is clicked', () => {
    const component = shallow(<RemoveButton {...defaultProps} />);

    component.find('button').simulate('click');
    expect(mockOnRemove).toBeCalledWith(defaultProps.targetFieldNumber);
  });
});
