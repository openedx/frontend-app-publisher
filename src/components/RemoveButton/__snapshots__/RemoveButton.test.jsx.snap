// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveButton renders correctly 1`] = `
<button
  aria-label="Test Label"
  className="close float-none d-flex justify-content-center align-items-center"
  onClick={[Function]}
  title="Test Label"
  type="button"
>
  <SvgClose
    id="remove-field"
  />
</button>
`;
